Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -121,119 +121,214 @@ uint64_t AODProducerWorkflowDPL::getTFNumber(uint64_t firstVtxGlBC, int runNumbe
+                          truncateFloatFraction(length, mTrackSignal),
+                          truncateFloatFraction(tofExpMom, mTrack1Pt),
+                          truncateFloatFraction(trackEtaEMCAL, mTrackPosEMCAL),
+                          truncateFloatFraction(trackPhiEMCAL, mTrackPosEMCAL));
+      }
+      if (trackType == GIndex::Source::ITSTPC) {
+        auto& track = tracksITSTPC[vTracksIdx[i][0]];
+        collisionID = vCollRefsTPCITS[vTracksIdx[i][0]];
+        tracksCursor(0,
+                     collisionID,
+                     trackType,
+                     truncateFloatFraction(track.getX(), mTrackX),
+                     truncateFloatFraction(track.getAlpha(), mTrackAlpha),
+                     track.getY(),
+                     track.getZ(),
+                     truncateFloatFraction(track.getSnp(), mTrackSnp),
+                     truncateFloatFraction(track.getTgl(), mTrackTgl),
+                     truncateFloatFraction(track.getQ2Pt(), mTrack1Pt));
+        tracksCovCursor(0,
+                        truncateFloatFraction(TMath::Sqrt(track.getSigmaY2()), mTrackCovDiag),
+                        truncateFloatFraction(TMath::Sqrt(track.getSigmaZ2()), mTrackCovDiag),
+                        truncateFloatFraction(TMath::Sqrt(track.getSigmaSnp2()), mTrackCovDiag),
+                        truncateFloatFraction(TMath::Sqrt(track.getSigmaTgl2()), mTrackCovDiag),
+                        truncateFloatFraction(TMath::Sqrt(track.getSigma1Pt2()), mTrackCovDiag),
+                        (Char_t)(128. * track.getSigmaZY() / track.getSigmaZ2() / track.getSigmaY2()),
+                        (Char_t)(128. * track.getSigmaSnpY() / track.getSigmaSnp2() / track.getSigmaY2()),
+                        (Char_t)(128. * track.getSigmaSnpZ() / track.getSigmaSnp2() / track.getSigmaZ2()),
+                        (Char_t)(128. * track.getSigmaTglY() / track.getSigmaTgl2() / track.getSigmaY2()),
+                        (Char_t)(128. * track.getSigmaTglZ() / track.getSigmaTgl2() / track.getSigmaZ2()),
+                        (Char_t)(128. * track.getSigmaTglSnp() / track.getSigmaTgl2() / track.getSigmaSnp2()),
+                        (Char_t)(128. * track.getSigma1PtY() / track.getSigma1Pt2() / track.getSigmaY2()),
+                        (Char_t)(128. * track.getSigma1PtZ() / track.getSigma1Pt2() / track.getSigmaZ2()),
+                        (Char_t)(128. * track.getSigma1PtSnp() / track.getSigma1Pt2() / track.getSigmaSnp2()),
+                        (Char_t)(128. * track.getSigma1PtTgl() / track.getSigma1Pt2() / track.getSigmaTgl2()));
+        tracksExtraCursor(0,
+                          truncateFloatFraction(tpcInnerParam, mTrack1Pt),
+                          flags,
+                          itsClusterMap,
+                          tpcNClsFindable,
+                          tpcNClsFindableMinusFound,
+                          tpcNClsFindableMinusCrossedRows,
+                          tpcNClsShared,
```

Comment:
same here: these variable will  have the values of last unrelated track which set it.

Reply:
Thank you for noticing, I have fixed this in the latest commit